lib/rate-limit.js

```diff
@@ -0,0 +1,85 @@
+var Promise = require('promise'),
+    debug = require('debug')('pulldasher:rate-limit');
```

Comment:
The only *rate limiting* part of the code that isn't specific to our use case is the last 25 lines of this file (the rest deals with github headers, choke points, ...).

Never the less, every one I looked at either through away extra calls (i.e. make 10 calls == only 2 get through) and / or had a fixed time interval on creation. We want our interval to be flexible and respond to the amount of time / requests remaining.